packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -100,22 +108,359 @@ class VMInfoList extends StatelessWidget {
+          'Retained by a GC root of type ${retainingPath.gcRootType ?? '<unknown>'}',
+          style: Theme.of(context).fixedFontStyle,
+        ),
+      ],
+    )
+  ];
+
+  final retainingObjectsRows = _prettyRows(context, retainingObjects);
+
+  return <Widget>[...retainingObjectsRows];
+}
+
+/// Describes the given RetainingObject [object] and its parentListIndex,
+/// parentMapKey, and parentField where applicable.
+String _retainingObjectDescription(RetainingObject object) {
+  if (object.parentListIndex != null) {
+    final ref = object.value as InstanceRef;
+    return 'Retained by element [${object.parentListIndex}] of ${ref.classRef?.name ?? '<parentListName>'}';
+  }
+
+  if (object.parentMapKey != null) {
+    final ref = object.value as InstanceRef;
+    return 'Retained by element [${object.parentMapKey}] of ${ref.classRef?.name ?? '<parentMapName>'}';
+  }
+
+  String description = 'Retained by';
+
+  if (object.parentField != null) {
+    description += ' ${object.parentField} of ';
+  }
+
+  if (object.value is FieldRef) {
+    final ref = object.value as FieldRef;
+    description +=
+        ' ${ref.declaredType?.name ?? 'Field'} ${ref.name} of ${_ownerName(ref.owner) ?? '<Owner>'}';
+  } else if (object.value is FuncRef) {
+    final ref = object.value as FuncRef;
+    description += ' ${_ownerName(ref.owner) ?? '<Owner>'}.${ref.name}';
+  } else {
+    description += ' ${_objectName(object.value)}';
+  }
+
+  return description;
+}
+
+/// An expandable list to display the inbound references for a given
+/// instance of InboundReferences.
+class InboundReferencesWidget extends StatelessWidget {
+  const InboundReferencesWidget({
+    required this.fetching,
+    required this.inboundReferences,
+    this.onExpanded,
+  });
+
+  final ValueListenable<bool> fetching;
+  final ValueListenable<InboundReferences?> inboundReferences;
+  final void Function(bool)? onExpanded;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<bool>(
+      valueListenable: fetching,
+      builder: (context, fetching, child) {
+        return VmExpansionTile(
+          title: 'Inbound References',
+          onExpanded: onExpanded,
+          children: inboundReferences.value == null
+              ? [
+                  SizedBox.fromSize(
+                    size: const Size.fromHeight(
+                      2 * (defaultIconSizeBeforeScaling + denseSpacing),
+                    ),
+                    child: const CenteredCircularProgressIndicator(),
+                  )
+                ]
+              : inboundReferencesList(
+                  context,
+                  inboundReferences.value!,
+                ),
+        );
+      },
+    );
+  }
+}
+
+/// Returns a list of Widgets that will be the rows in the VmExpansionTile
+/// for InboundReferencesWidget.
+List<Widget> inboundReferencesList(
+  BuildContext context,
+  InboundReferences inboundRefs,
+) {
+  int count = -1;
```

Comment:
Nit: set this to zero and then increment it at the bottom of the loop.

Reply:
Thanks!